awx/api/serializers.py

```diff
@@ -5368,6 +5374,11 @@ class Meta:
         fields = ('id', 'hostname', 'node_type', 'node_state', 'enabled')
 
 
+class PeersSerializer(serializers.StringRelatedField):
+    def to_internal_value(self, value):
+        return Instance.objects.get(hostname=value)
```

Comment:
I'm curious why this was done, instead of making `InstanceSerializer.peers` something like the `source` and `target` fields on `InstanceLinkSerializer`.  It seems to me like doing it this way will result in N+1 query behavior.